File: src/2018/day_01.mjs
import * as fs from 'node:fs';

const inputData = fs.readFileSync('input.txt', 'utf-8').trim().split('\n').map(n => parseInt(n));

console.log(`Part 1: ${inputData.reduce((acc, cur) => acc + cur, 0)}`);

const findRepeatingFrequency = () => {
  const frequencyHistory = new Set();
  let i = 0;
  let frequency = 0;

  while (true) {
    frequency += inputData[i];
    if (frequencyHistory.has(frequency)) return frequency;
    frequencyHistory.add(frequency);
    i = (i + 1) % inputData.length;
  }
};

console.log(`Part 2: ${findRepeatingFrequency()}`);

File: 2022/day_01.dart
import 'dart:io';

void main() {
  List<List<int>> inputData = File('input.txt')
      .readAsStringSync()
      .trim()
      .split('\n\n')
      .map((elf) => elf.split('\n').map((cal) => int.parse(cal, radix: 10)).toList())
      .toList();

  List<int> totalCalories = inputData
      .map((elf) => elf.reduce((acc, cur) => acc + cur))
      .toList();

  totalCalories.sort((a, b) => b - a);

  print('Part 1: ${totalCalories[0]}');
  print('Part 2: ${totalCalories.sublist(0, 3).reduce((acc, cur) => acc + cur)}');
}
